config/configtls/configtls.go

```diff
@@ -189,11 +189,25 @@ func (r *certReloader) GetCertificate() (*tls.Certificate, error) {
 	return r.cert, nil
 }
 
-func (c Config) Validate() error {
+func (c Config) validate() error {
```

Comment:
I just realized this is technically a breaking API change, which we can't have because `configtls` is stable; sorry about all the flip-flopping...

```suggestion
func (c Config) Validate() error {
```

(This means that we might end up failing validation of fields which are ignored because of `ClientConfig.Insecure`, but I think it should be fine as long as the case where all other fields are empty continues to work.)